rs/tests/httpbin-rs/src/main.rs

```diff
@@ -89,6 +89,8 @@ async fn redirect_handler(Path(n): Path<u64>) -> impl IntoResponse {
 
 /// Builds the response body using the request
 async fn anything_handler(method: Method, headers: HeaderMap, body: String) -> Vec<u8> {
+    // TODO: Do not use a map? Otherwise we lose duplicate headers, making the
+    // check in the tests to never fail.
```

Comment:
If the canister HTTPS outcalls adapter uses a HeaderMap, too, then it'd never send duplicated headers anyway.